utils.sh

```diff
@@ -343,15 +344,23 @@ function generate_auth_template {
         IRONIC_USER=$((oc -n openshift-machine-api  get secret/metal3-ironic-password -o template --template '{{.data.username}}' || echo "") | base64 -d)
         IRONIC_PASSWORD=$((oc -n openshift-machine-api  get secret/metal3-ironic-password -o template --template '{{.data.password}}' || echo "") | base64 -d)
         IRONIC_CREDS="$IRONIC_USER:$IRONIC_PASSWORD"
-        INSPECTOR_USER=$((oc -n openshift-machine-api  get secret/metal3-ironic-inspector-password -o template --template '{{.data.username}}' || echo "") | base64 -d)
-        INSPECTOR_PASSWORD=$((oc -n openshift-machine-api  get secret/metal3-ironic-inspector-password -o template --template '{{.data.password}}' || echo "") | base64 -d)
-        INSPECTOR_CREDS="$INSPECTOR_USER:$INSPECTOR_PASSWORD"
+        if [[ "$OCP_VERSIONS_INSPECTOR" == *"$VERSION"* ]]; then
+            INSPECTOR_USER=$((oc -n openshift-machine-api  get secret/metal3-ironic-inspector-password -o template --template '{{.data.username}}' || echo "") | base64 -d)
+            INSPECTOR_PASSWORD=$((oc -n openshift-machine-api  get secret/metal3-ironic-inspector-password -o template --template '{{.data.password}}' || echo "") | base64 -d)
+            INSPECTOR_CREDS="$INSPECTOR_USER:$INSPECTOR_PASSWORD"
+        else
+            INSPECTOR_CREDS=""
```

Comment:
we can probably remove this else statement, INSPECTOR_CREDS is not needed if we don't use inspector

Reply:
Something is causing the jobs to fail ( [example](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift-metal3_dev-scripts/1683/pull-ci-openshift-metal3-dev-scripts-master-e2e-metal-ipi-bm/1845831262900588544#1:build-log.txt%3A19931-19976) ), can this be related.

Reply:
yep, just tested this locally and I think we should remove -inspector-basic-auth in L362

Reply:
if we can't change the arguments conditionally in metal3-templater.go the value should be -ironic-basic-auth=":" in case of inspector absence

Reply:
Updated.

I roved the else branch.  I made the `-inspector-basic-auth` flag optional in the templater and then I removed the flag at the call site.